hosts/binarycache/configuration.nix

```diff
@@ -54,13 +55,28 @@
   };
 
   services.nginx = {
+    package = pkgs.nginxStable.override {
+      modules = [pkgs.nginxModules.zstd];
+    };
+    recommendedZstdSettings = true;
     virtualHosts = {
       "cache.vedenemo.dev" = {
         enableACME = true;
         forceSSL = true;
         default = true;
         locations."/" = {
           proxyPass = "http://${config.services.nix-serve.bindAddress}:${toString config.services.nix-serve.port}";
+          extraConfig = ''
+            proxy_set_header Host $host;
+            proxy_redirect http:// https://;
+            proxy_http_version 1.1;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection $connection_upgrade;
```

Comment:
There are options that set these:

- `services.nginx.virtualHosts.<name>.locations.<name>.recommendedProxySettings` 
  https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix#L228
  https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix#L97
  This one is actually already set globally in our nginx module.

- `services.nginx.virtualHosts.<name>.locations.<name>.proxyWebsockets`
  https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix#L439

Reply:
Mostly I've just adapted the settings from the source. It's really hard to see what is already set and what is not by just looking at the files in editor. Wish there was a tool helping with this. Anyway, good catch, I'll make some changes.